ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt

```diff
@@ -206,4 +213,13 @@ class ArgumentListWrappingRule : Rule("argument-list-wrapping") {
         }
         return null
     }
+
+    private fun ASTNode.isOnSameLineAsIfKeyword(): Boolean {
+        val containerNode = psi.getStrictParentOfType<KtContainerNode>() ?: return false
+        return when (val parent = containerNode.parent) {
```

Comment:
But what about `KtWhenExpression`?

Reply:
We have a test for `when`.
https://github.com/t-kameyama/ktlint/blob/74656c35796b2541335419ba55cc9d5ba2ec54fd/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRuleTest.kt#L366-L384

In when condition, intdent is 4.

Reply:
There is a test for `when`.
https://github.com/t-kameyama/ktlint/blob/74656c35796b2541335419ba55cc9d5ba2ec54fd/ktlint-ruleset-experimental/src/test/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRuleTest.kt#L366-L384

In `when` condition, intdent is 4.

Reply:
But shouldn't it be
```Kotlin
when(foo(
        i
    )
) { ... }
```
or am I missing something?

Reply:
IntelliJ formats it as follows:

```kotlin
fun foo(i: Int) = true

fun test(i: Int) {
    when (foo(
            i
    )) {
        true -> println(1)
        false -> println(2)
    }
}

```

to

```kotlin
fun foo(i: Int) = true

fun test(i: Int) {
    when (foo(
        i
    )) {
        true -> println(1)
        false -> println(2)
    }
}
```

Reply:
gotcha, thanks for clarificaiton